onnxruntime/test/providers/cpu/reduction/reduction_ops_test.cc

```diff
@@ -3275,11 +3553,11 @@ TEST(ReductionOpTest, ReduceDimWithZero1) {
   // reduce on all axes keeping dims. should allow the 0 to be the reduced value
   OpTester test("ReduceSum", 10);
   test.AddAttribute("keepdims", int64_t(1));
-  test.AddShapeToTensorData(true, 1);  // make second dim symbolic so that we don't break during shape inferencing
   test.AddInput<float>("data", {3, 0, 2}, {});
-  test.AddOutput<float>("reduced", {1, 0, 1}, {});
+  test.AddOutput<float>("reduced", {1, 1, 1}, {0.0f});
```

Comment:
Based on the test name, this line should not be changed.

Reply:
this is due to https://github.com/onnx/onnx/pull/5568 where reduce on empty sets was clarified. In this test case, the behavior is changed - If the empty dimension is also the dimension being reduced, that dimension changes to 1 and the value of reduced is 0. The test is to reduce on all axes which include axis 1 which has 0 dimension. So the shape: (3, 0, 2) => (1, 1, 1). And its value is 0. "ReduceDimWithZero1" is still a proper name. See for example: https://github.com/onnx/onnx/blob/b60f69412abb5393ab819b936b473f83867f6c87/onnx/backend/test/case/node/reducesum.py#L199